### PR TITLE
[2.1] Fix IPNetwork.Contains for prefixes which are not at start of subnet range.

### LINKED ISSUE #6674: Inconsistent IPNetwork.Contains behavior for prefixes which are not at start of subnet range.
### Describe the bug

In AspNetCore/src/Middleware/HttpOverrides/src/IPNetwork.cs

The code below assumes that the prefix will have all bits in position > mask length set to 0. I do not believe that this a requirement for CIDR notation. If it is, it is not applied consistently since the routine will break if the mask is == 0 at a given byte.

            for (int i = 0; i < PrefixBytes.Length && Mask[i] != 0; i++)
            {
                if (PrefixBytes[i] != (addressBytes[i] & Mask[i]))
                {
                    return false;
                }
            }

### To Reproduce
Create instance with Prefix/Prefix Length 192.168.0.1/31.
Contains returns false for 192.168.0.0.

### Expected behavior
Expectation is that 192.168.0.1/31 would match 192.168.0.0

### PR DESCRIPTION
**Pull request by adityamandaleeka**
# [2.1] Fix IPNetwork.Contains for prefixes which are not at start of subnet range

## Description

The code in IPNetwork.Contains was incorrectly handling certain cases because the way subnet membership comparison code was incorrect.

This is a backport of 3d009158 to 2.1.

6.0 PR: https://github.com/dotnet/aspnetcore/pull/31573

Fixes #6674 in 2.1

## Customer Impact

The result of the `Contains` method could be incorrect.

## Regression?

- [ ] Yes
- [x] No

## Risk

- [ ] High
- [ ] Medium
- [x] Low

Simple correctness change.

## Verification

- [x] Manual (required)
- [x] Automated

## Packaging changes reviewed?

- [ ] Yes
- [ ] No
- [x] N/A

----

## When servicing release/2.1

- [x] Make necessary changes in eng/PatchConfig.props

### CODE REVIEW

**Comment by adityamandaleeka** [comment]
I'll add the template and update patchconfig once validation is complete.

**Review by Tratcher — APPROVED** [review]

**Comment by adityamandaleeka** [comment]
@BrennanConroy Can I get another review please?
@wtgodbe Just waiting for tactics approval now.

**Review by wtgodbe — APPROVED** [review]

File: eng/PatchConfig.props
Comment (BrennanConroy) [inline eng/PatchConfig.props:142]: I assume this should be in a new section for 2.1.41?

**Review by BrennanConroy — APPROVED** [review]
pending patch.config comment

File: eng/PatchConfig.props
Comment (adityamandaleeka) [reply eng/PatchConfig.props:142]: @wtgodbe is going to handle that.

**Pull request merged**
